torch_em/data/datasets/pannuke.py

```diff
@@ -0,0 +1,149 @@
+import os
+import h5py
+import vigra
+import shutil
+import numpy as np
+from glob import glob
+
+import torch_em
+from torch_em.util.debug import check_loader
+from torch_em.data.datasets import util
+
+
+# PanNuke Dataset - https://warwick.ac.uk/fac/cross_fac/tia/data/pannuke
+URLS = {
+    "fold_1": "https://warwick.ac.uk/fac/cross_fac/tia/data/pannuke/fold_1.zip",
+    "fold_2": "https://warwick.ac.uk/fac/cross_fac/tia/data/pannuke/fold_2.zip",
+    "fold_3": "https://warwick.ac.uk/fac/cross_fac/tia/data/pannuke/fold_3.zip"
+}
+
+
+# TODO
+CHECKSUM = None
+
+
+def _download_pannuke_dataset(path, download, folds):
+    os.makedirs(path, exist_ok=True)
+
+    checksum = CHECKSUM
+
+    for tmp_fold in folds:
+        if os.path.exists(os.path.join(path, f"pannuke_{tmp_fold}.h5")):
+            return
+
+        util.download_source(os.path.join(path, f"{tmp_fold}.zip"), URLS[f"{tmp_fold}"], download, checksum)
+
+        print(f"Unzipping the PanNuke dataset in {tmp_fold} directories...")
+        util.unzip(os.path.join(path, f"{tmp_fold}.zip"), os.path.join(path, f"{tmp_fold}"), True)
+
+
+def _convert_to_hdf5(path, folds):
+    for tmp_fold in folds:
+        if os.path.exists(os.path.join(path, f"pannuke_{tmp_fold}.h5")):
+            return
+
+        print(f"Converting the {tmp_fold} into h5 file format...")
+        tmp_name = tmp_fold.split("_")[0] + tmp_fold.split("_")[1]  # name of a particular sub-directory (per fold)
+        with h5py.File(os.path.join(path, f"pannuke_{tmp_fold}.h5"), "w") as f:
+            img_path = glob(os.path.join(path, tmp_fold, "*", "images", tmp_name, "images.npy"))[0]
+            gt_path = glob(os.path.join(path, tmp_fold, "*", "masks", tmp_name, "masks.npy"))[0]
```

Comment:
You only take a single of the files here? Is this on purpose?
I would have assumed that the code should look something like this instead:
```python
img_paths = sorted(glob(..., "images.npy))  # imported to do sorted here, so that you get the same order for masks and images
gt_paths = sorted(glob(..., "masks.npy))  # imported to do sorted here, so that you get the same order for masks and images
for i, (img_path, gt_path) in enumerate(zip(img_paths, gt_paths)):
  img = np.load(img_path)
  gt = np.load(gt_path)
  assert img,shape == gt.shape  # or similar; but make sure that the shapes match, you might need to account for a different number of channels
  out_path = os.path.join(..., f"pannuke_{tmp_fold}_{i}.h5"
  with h5py.File(out_path, "w") as f:
      ...
```
That way you would use all the images in a given fold

Reply:
Ah, it's to provide the option to (only) use specific folds of the dataset (there are 3 in total now) (I took the inspiration for this from cremi). Do you think we should do the download and h5 conversion at once for all the folds already?

Reply:
> Ah, it's to provide the option to (only) use specific folds of the dataset (there are 3 in total now) (I took the inspiration for this from cremi).

That's not what I mean. Having the separate folds is ok. But with the `glob` here and then accessing only the first it looks like you're just selecting one image.
```python
glob(os.path.join(path, tmp_fold, "*", "images", tmp_name, "images.npy"))[0]
```
Maybe I am also understanding the data organization wrong and it's many images stacked. But if that is the case we should save them in separate files to match the data organization expected by torch_em better.